src/requests/requests.cpp

```diff
@@ -113,4 +113,26 @@ bool CreateAndDownloadFile(const std::string &url, const std::string &path, int
   return true;
 }
 
+auto DownloadToStream(char const *url) -> std::stringstream {
+  constexpr auto WriteCallback = [](char *ptr, size_t size, size_t nmemb, std::stringstream *stream) -> size_t {
+    size_t totalSize = size * nmemb;
+    stream->write(ptr, totalSize);
+    return totalSize;
+  };
+
+  auto ss = std::stringstream{};
+
+  auto *curl_handle{curl_easy_init()};
+  curl_easy_setopt(curl_handle, CURLOPT_URL, url);
+  curl_easy_setopt(curl_handle, CURLOPT_WRITEFUNCTION, +WriteCallback);
+  curl_easy_setopt(curl_handle, CURLOPT_WRITEDATA, &ss);
+
+  auto const res = curl_easy_perform(curl_handle);
+  curl_easy_cleanup(curl_handle);
+  if (res != CURLE_OK) {
+    throw 1;  // TODO: A sensible exception (problem: BasicException in utils)
```

Comment:
What about passing `ss` by ref and return bool to indicate the status of this function call (just to be aligned with the rest of the implementation here)?

Reply:
yes that work, pushed that change up